From: Oliver Giles <oliver_g@radwin.com>
Date: Wed, 28 Aug 2019 17:29:48 +0300
Subject: [PATCH] folly-deps: exclude linked libraries from INTERFACE

The folly-deps INTERFACE library exposed library paths
found with find_library to downstream packages via the
INTERFACE_LINK_LIBRARIES and INTERFACE_INCLUDE_DIRECTORIES
properties. This is poor design, as pointed out in the
CMake documentation for target_include_directories,
because it hard-codes include directory paths and link
library paths to those where folly was built.

---
 CMake/folly-deps.cmake | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 989259a87..ecbc6a5c4 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -307,9 +307,11 @@ endif()
 target_link_libraries(folly_deps INTERFACE fmt::fmt)

 list(REMOVE_DUPLICATES FOLLY_INCLUDE_DIRECTORIES)
-target_include_directories(folly_deps INTERFACE ${FOLLY_INCLUDE_DIRECTORIES})
+#target_include_directories(folly_deps INTERFACE ${FOLLY_INCLUDE_DIRECTORIES})
 target_link_libraries(folly_deps INTERFACE
-  ${FOLLY_LINK_LIBRARIES}
-  ${FOLLY_SHINY_DEPENDENCIES}
+  #${FOLLY_LINK_LIBRARIES}
+  #${FOLLY_SHINY_DEPENDENCIES}
+  $<BUILD_INTERFACE:${FOLLY_LINK_LIBRARIES}>
+  $<INSTALL_INTERFACE:${FOLLY_SHINY_DEPENDENCIES}>
   ${FOLLY_ASAN_FLAGS}
 )
--
2.30.2
